master/networking/ip-autodetection.md

```diff
@@ -0,0 +1,204 @@
+
+- **IPv4**
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "<autodetection-method>"}]}]}}}}'
+  ```
+
+- **IPv6**
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP6_AUTODETECTION_METHOD", "value": "<autodetection-method>"}]}]}}}}'
+  ```
+
+Where auto-detection methods are based on:
+
+- **IP or domain name**
+
+  A reachable destination (IP address or domain). For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "can-reach=8.8.8.8"}]}]}}}}'
+
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "can-reach=www.google.com"}]}]}}}}'
+  ```
+
+- **Including matching interfaces**
+
+  A regular expression in golang syntax that includes interfaces that match. For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "interface=eth.*"}]}]}}}}'
+  ```
+
+- **Excluding matching interfaces**
+
+  A regular expression in golang syntax that excludes interfaces that match. For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "skip-interface=eth.*"}]}]}}}}'
+  ```
+
+#### Manually configure IP address and subnet for a node
+
+In the following scenarios, you may want to configure a specific IP and subnet:
+
+- Hosts with multiple external interfaces
+- Host interfaces with multiple IP addresses
+- Changes to cross subnet packet encapsulation
+- Changes to host IP address
+
+You can configure specific IP address and subnet for a node using environment variables or by updating the [Node resource]({{site.baseurl}}/{{page.version}}/reference/resources/node).
+
+##### Configure IP and subnet using environment variables
+
+To configure IP and subnet values using environment variables, use a `kubectl patch` command. For example:
+
+```
+kubectl patch ds -n kube-system calico-node --patch \
+    '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP", "value": "10.0.2.10/24"}, {"name": "IP6", "value": "fd80:24e2:f998:72d6::/120"}]}]}}}}'
+```
+
+Where:
+
+- IP4 flag: --ip
```

Comment:
I think these flags don't make sense here any more? 

@lmm does that sound right?

Reply:
Oops, yes fixed that here d2a927db52d6cd4ea462977ce4e4f833c4af1f3a